### PR TITLE
[2.3.2.r1.4] Migrate MSM8956 to msm_cdc_pinctrl, remove msm-audio-pinctrl driver

### PR DESCRIPTION
**Pull request by kholk**
This patchset brings MSM8956 up to date by migrating it to the new style
msm_cdc_pinctrl instead of relying on the old and long deprecated gpioset
based driver msm-audio-pinctrl (and also removes it).

Also, there's a correction for the audio pins setup which, for the related PMIC
GPIOs, was _really_ wrong.

This PR depends on #1981 
Tested on SoMC Loire Suzu RoW (no regressions seen).

### CODE REVIEW

File: techpack/audio/asoc/msm8952.h [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+ * it under the terms of the GNU General Public License version 2 and
+ * only version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#ifndef __MSM8953_H
```
Comment (bartcubbins) [inline techpack/audio/asoc/msm8952.h:13]: ping @kholk

Comment (kholk) [reply techpack/audio/asoc/msm8952.h:13]: thanks, but don't worry, that's fine :D

Comment (bartcubbins) [reply techpack/audio/asoc/msm8952.h:13]: oops, yes:)
https://source.codeaurora.org/quic/la/platform/vendor/opensource/audio-kernel/tree/asoc/msm8952.h?h=LA.UM.7.8.r3-01400-SDM710.0#n13

**Comment by oshmoun** [comment]
works nicely so far
music playback is ok
incall audio is fine even using bt

**Comment by kholk** [comment]
I've left my suzu up and running to test deep sleep effects.
The phone is still outputting sound like normal. 

Green light from me!

**Pull request merged**